odoo/addons/base/models/res_partner.py

```diff
@@ -598,6 +598,13 @@ def create(self, vals_list):
             partner._handle_first_contact_creation()
         return partners
 
+    @api.constrains('email')
+    def _check_email_coding(self):
```

Comment:
A constraint means more defensive code (code that serves no purpose except checking for one specific error) + more complexity for the users. Many wouldn't even understand that error message.

I think a good alternative is to fix the computation of `email_formatted` to make it more lenient with the contents of the `email` field, e.g. via `tools.email_split()`. This is what we do in many other places where we need to extract a valid email from a field.

(In v13 onwards this could be further simplified to use `tools.email_normalize()`

```patch
diff --git odoo/addons/base/models/res_partner.py odoo/addons/base/models/res_partner.py
index ac28c7a7467..fe8cc80bc11 100644
--- odoo/addons/base/models/res_partner.py
+++ odoo/addons/base/models/res_partner.py
@@ -404,7 +404,9 @@ class Partner(models.Model):
     def _compute_email_formatted(self):
         for partner in self:
             if partner.email:
-                partner.email_formatted = tools.formataddr((partner.name or u"False", partner.email or u"False"))
+                emails = tools.email_split(partner.email)
+                email = emails[0] if emails else "False"
+                partner.email_formatted = tools.formataddr((partner.name or u"False", email))
             else:
                 partner.email_formatted = ''

```
This shouldn't break anything, and will make the system more robust.

Some remaining concerns:
- There are other similar places that may need a similar fix, like [this](https://github.com/odoo/odoo/blob/df9de06d260b313f0db61f47b7ebdcdd1c9c6cf6/addons/mail/models/mail_mail.py#L204)
- To make this perfect you would also need to add tests to cover it, maybe [here](https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/tests/test_res_partner.py#L1).

The people you want reviewing this are probably @tde-banana-odoo and @Julien00859.

PS: @Julien00859 in `master` `format_addr()` would perhaps not crash anymore, but would still produce unexpected results.